moped-editor/src/components/FeedbackSnackbar.js

```diff
@@ -1,33 +1,87 @@
-import React from "react";
-import { Alert } from '@mui/material';
+import React, { useCallback, useState } from "react";
+import { Alert } from "@mui/material";
 import { Snackbar } from "@mui/material";
 
 /**
- * A project summary snackbar
- * @param {Object} snackbarState - The message you would like to see
- * @param {function} handleSnackbar - The function to update state
- * @param {Number} timeOut - Time out in milliseconds (default: 5000 milliseconds--for five seconds)
+ * Custom hook to manage snackbar state
+ * @returns {Object} - snackbarState, handleSnackbar, handleSnackbarClose
+ */
+export const useFeedbackSnackbar = () => {
+  /**
+   * State for the snackbar
+   * @property {boolean} open - Whether the snackbar is open
+   * @property {String} message - The message to display in the snackbar
+   * @property {String} severity - The MUI Alert severity of the snackbar ("success", "error", "warning", "info")
+   */
+  const [snackbarState, setSnackbarState] = useState({
+    open: false,
+    message: "",
+    severity: "",
+  });
+
+  /**
+   * Wrapper around snackbar state setter
+   * @param {boolean} open - The new state of open
+   * @param {String} message - The message for the snackbar
+   * @param {String} severity - The MUI Alert severity of the snackbar ("success", "error", "warning", "info")
+   * @param {Object} error - The error to be displayed and logged
+   */
+  const handleSnackbar = useCallback(
+    (open, message, severity, error) => {
+      // if there is an error, render error message,
+      // otherwise, render success message
+      if (error) {
+        setSnackbarState({
+          open: open,
+          message: `${message}. Refresh the page to try again.`,
+          severity: severity,
+        });
+        console.error(error);
+      } else {
+        setSnackbarState({
+          open: open,
+          message: message,
+          severity: severity,
+        });
+      }
+    },
+    [setSnackbarState]
+  );
+
+  /**
+   * Callback to reset the snackbar state on snackbar close
+   */
+  const handleSnackbarClose = (_, reason) => {
```

Comment:
I was unfamiliar with this syntax and there are some places where i have definitely included variables I was ignoring (I am thinking `event` for example) and this seems much cleaner

Reply:
cool, i'm on the fence really. I can totally see the value of including `event` so we know it is available in the future. We don't enforce a lint rule about it either so either way flys! 😄